src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp

```diff
@@ -244,7 +246,21 @@ void AzureKinectFrameProvider::SetTransformedBodyMaskBuffer(uint16_t* transforme
 
 IFrameProvider::ProviderType AzureKinectFrameProvider::GetProviderType()
 {
-    return IFrameProvider::ProviderType::AzureKinect;
+    switch (depthCameraMode)
+    {
+    case K4A_DEPTH_MODE_OFF:
+
+        return IFrameProvider::ProviderType::AzureKinect_DepthCamera_Off;
+
+    case K4A_DEPTH_MODE_NFOV_UNBINNED:
+
+        return IFrameProvider::ProviderType::AzureKinect_DepthCamera_NFOV;
+
+    case K4A_DEPTH_MODE_WFOV_2X2BINNED:
+
+        return IFrameProvider::ProviderType::AzureKinect_DepthCamera_WFOV;
+
+    }
```

Comment:
There should be a default block here that returns something (maybe the default case is returning the AzureKinect_DepthCamera_Off)

Reply:
Makes sense since that is the default depth mode

---
In reply to: [378390353](https://github.com/matthejo/MixedReality-SpectatorView/pull/6#discussion_r378390353) [](ancestors = 378390353)